cli/cli.ts

```diff
@@ -3367,6 +3377,84 @@ export function extractAsync(filename: string) {
         })
 }
 
+export function preCacheHexAsync() {
+    let trgInfo = readLocalPxTarget();
+
+    if (!trgInfo) {
+        console.log("pxtarget.json not found; make sure you are in a valid PXT target directory");
+        return Promise.resolve();
+    }
+
+    // Extract the target's default project to disk
+    if (!trgInfo.blocksprj) {
+        console.log("Could not find default project 'blocksprj' in pxtarget.json");
+        return Promise.resolve();
+    }
+
+    let projectsRoot = "projects";
+    let projectInitialName = "_precache_hex";
+    let projectName = projectInitialName;
+    let projectPath = path.join(projectsRoot, projectName);
+    let counter = 2;
+
+    while (fs.existsSync(projectPath)) {
```

Comment:
isn't this going to a new project each time you need to compile the precache? If this project folder was hosted under built/precache, we could simply nuke any previous files.